superset/dashboards/dao.py

```diff
@@ -40,11 +40,22 @@ class DashboardDAO(BaseDAO):
     base_filter = DashboardAccessFilter
 
     @staticmethod
-    def get_by_id_or_slug(id_or_slug: str) -> Dashboard:
-        dashboard = Dashboard.get(id_or_slug)
```

Comment:
Doesn't FAB fetch all the relationships as well, i.e., why do we need to explicitly compose a query which joins all the related models?

Reply:
it does but the get item endpoint for dashboards is custom on Superset because we need to fetch a dashboard by `id` or `slug`, so on this case we need to custom develop. FAB `ModelRestApi` on get item endpoints will get item by the table primary key